src/main/scala/tilelink/RegisterRouter.scala

```diff
@@ -81,19 +85,52 @@ case class TLRegisterNode(
     bundleIn.c.ready := Bool(true)
     bundleIn.e.ready := Bool(true)
 
+    genRegDescsJson(mapping: _*)
+  }
+
+
+  def genRegDescsJson(mapping: RegField.Map*) : Unit = {
     // Dump out the register map for documentation purposes.
     val base = address.head.base
     val baseHex = s"0x${base.toInt.toHexString}"
     val name = s"deviceAt${baseHex}" //TODO: It would be better to name this other than "Device at ...."
-    val json = RegMappingAnnotation.serialize(base, name, mapping:_*)
+    val json = GenRegDescJson.serialize(base, name, mapping:_*)
     var suffix = 0
-    while( ElaborationArtefacts.contains(s"${baseHex}.${suffix}.regmap.json")){
+    while( ElaborationArtefacts.contains(s"${baseHex}.${suffix}.regmap.json")) {
       suffix = suffix + 1
     }
     ElaborationArtefacts.add(s"${baseHex}.${suffix}.regmap.json", json)
+
+    /* GetMeMyModule is a hack to get the current module during elaboration.
```

Comment:
I think this comment should go in the `GetMyModule` file itself.

Reply:
This comment serves as a reminder to change this code to use the new def in Chisel and remove the hack. Comment updated